contracts/src/token/erc20/extensions/flash_mint.rs

```diff
@@ -0,0 +1,397 @@
+            ) external returns (bytes32);
+        }
+    }
+}
+
+/// State of the [`Erc20FlashMint`] Contract.
+#[storage]
+pub struct Erc20FlashMint {
+    /// Fee applied when doing flash loans.
+    pub flash_fee_amount: StorageU256,
+    /// Receiver address of the flash fee.
+    pub flash_fee_receiver_address: StorageAddress,
+}
+
+/// Interface of the ERC-3156 Flash Lender, as defined in [ERC-3156].
+///
+/// [ERC-3156]: https://eips.ethereum.org/EIPS/eip-3156
+pub trait IErc3156FlashLender {
+    /// The error type associated to this trait implementation.
+    type Error: Into<alloc::vec::Vec<u8>>;
+
+    /// Returns the maximum amount of tokens available for loan.
+    ///
+    /// NOTE: This function does not consider any form of supply cap, so in case
+    /// it's used in a token with a cap like
+    /// [`crate::token::erc20::extensions::Capped`], make sure to override this
+    /// function to integrate the cap instead of [`U256::MAX`].
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `token` - The address of the token that is requested.
+    /// * `erc20` - Read access to an [`Erc20`] contract.
+    fn max_flash_loan(&self, token: Address, erc20: &Erc20) -> U256;
+
+    /// Returns the fee applied when doing flash loans.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `token` - The token to be flash loaned.
+    /// * `amount` - The amount of tokens to be loaned.
+    ///
+    /// # Errors
+    ///
+    /// If the token is not supported, then the error
+    /// [`Error::UnsupportedToken`] is returned.
+    fn flash_fee(
+        &self,
+        token: Address,
+        amount: U256,
+    ) -> Result<U256, Self::Error>;
+
+    /// Performs a flash loan.
+    ///
+    /// New tokens are minted and sent to the `receiver`, who is required to
+    /// implement the [`IERC3156FlashBorrower`] interface. By the end of the
+    /// flash loan, the receiver is expected to own value + fee tokens and have
+    /// them approved back to the token contract itself so they can be burned.
+    ///
+    /// Returns a boolean value indicating whether the operation succeeded.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// * `receiver` - The receiver of the flash loan. Should implement the
+    ///   [`IERC3156FlashBorrower::on_flash_loan`] interface.
+    /// * `token` - The token to be flash loaned. Only [`contract::address()`]
+    ///   is supported.
+    /// * `value` - The amount of tokens to be loaned.
+    /// * `data` - Arbitrary data that is passed to the receiver.
+    /// * `erc20` - Write access to an [`Erc20`] contract.
+    ///
+    /// # Errors
+    ///
+    /// If the `amount` is greater than the value returned by
+    /// [`IErc3156FlashLender::max_flash_loan`], then the error
+    /// [`Error::ExceededMaxLoan`] is returned. If `token` is not supported,
```

Comment:
Let's try to have a convention that each error starts with the next line, sth like this:
```
If <sth>, than the error <error> is returned.
If <sth>, than the error <error> is returned.
If <sth>, than the error <error> is returned.
```

Reply:
It seems my formatter messed this up

Reply:
We should probably use lists for errors (as we do write them out as lists). 

Currently the hover tooltip on VS Code displays errors as a single paragraph:
![image](https://github.com/user-attachments/assets/c149d6c2-2514-4013-a284-96445974bf42)
![image](https://github.com/user-attachments/assets/c3ddcc41-bc7d-4e60-bd86-08c8830e0e8c)

If we used lists it looks much more organized to me:
![image](https://github.com/user-attachments/assets/7b9a23e9-56ba-446f-8d44-691d05789116)
![image](https://github.com/user-attachments/assets/f34396b1-b817-430f-aa15-e2567387f833)

Reply:
Yeah. Definitely. Should look much better

Reply:
Much better.